app/models/pipeline_run.rb

```diff
@@ -1038,6 +1057,32 @@ def compile_stats_file
     save
   end
 
+  # Fetch the unmapped reads count from alignment stage then refined counts from
+  # assembly stage, as each becomes available. Prior to Dec 2019, the count was
+  # only fetched from alignment.
+  def fetch_unmapped_reads(
+    all_counts,
+    s3_path = "#{postprocess_output_s3_path}/#{ASSEMBLY_PREFIX}#{DAG_ANNOTATED_COUNT_BASENAME}"
+  )
+    unmapped_reads = nil
+    unidentified = all_counts.detect { |entry| entry.value?("unidentified_fasta") }
+    if unidentified
+      unmapped_reads = unidentified[:reads_after]
+
+      if supports_assembly? && finalized?
```

Comment:
I'm not sure the `finalized?` should be there. It seems possible for the file to exist even though the pipeline isn't done yet, and I think we'd want to display the value once it's ready.

Reply:
Yes I added that as an afterthought. I wanted to avoid the result monitor polling for a file that’s not there most of the time. Is there some state var that is closely coupled? Assembly done?

Reply:
Not that I know of, but you might be able to write an assembly_done method by checking whether pipeline run stage step 3 is complete. Will leave that up to you.